### PR TITLE
Add confirm_message to toggle-bool-switch

### PR DESCRIPTION
**Pull request by mario-amazing**
### Motivation / Background
Add new confirm_message feature to toggle-bool-switch: https://github.com/platanus/activeadmin_addons/issues/483

### CODE REVIEW

**Comment by mario-amazing** [comment]
@difernandez, could you check it?

File: app/javascript/activeadmin_addons/addons/toggle_bool.js [hunk 1]
```diff
@@ -2,6 +2,11 @@ var initializer = function() {
   $('.toggle-bool-switch').click(function(e) {
     var boolSwitch = $(e.target);
 
+    var confirmMessage = boolSwitch.data('confirm_message');
+    if (confirmMessage && !confirm(confirmMessage)) {
+      return false;
+    }
```
Comment (mpapis) [inline app/javascript/activeadmin_addons/addons/toggle_bool.js:8]:
```suggestion
    var value = boolSwitch.data('value');
    var confirmMessage = boolSwitch.data('confirm_message');
    if (confirmMessage && !confirm(confirmMessage)) {
      return false;
    }
    var confirmMessage = boolSwitch.data('confirm_on_message');
    if (confirmMessage && !value && !confirm(confirmMessage)) {
      return false;
    }
    var confirmMessage = boolSwitch.data('confirm_off_message');
    if (confirmMessage && value && !confirm(confirmMessage)) {
      return false;
    }
```

File: docs/toggle_bool.md [hunk 2]
```diff
@@ -39,6 +39,16 @@ The value update is done through the default update route, so you must check you
  `toggle_bool_column :paid, if: proc { |item| item.price.present? }`
  
  `toggle_bool_column :paid, unless: proc { |item| item.is_free? }`
+
+ ### Confirmation message
+
+ Optionally, you can provide a confirmation message to the user before updating the value.
+  
+ This is disabled by default, but can be enabled by adding the option `confirm_message`
```
Comment (mpapis) [inline docs/toggle_bool.md:47]:
```suggestion
 This is disabled by default, but can be enabled by adding a option:
 - `confirm_message` (on/off),
 - `confirm_on_message` (on),
 - `confirm_off_message` (off)
```

File: lib/activeadmin_addons/addons/toggle_bool_builder.rb [hunk 3]
```diff
@@ -24,7 +24,8 @@ def toggle
         'data-field' => attribute,
         'data-value' => data,
         'data-url' => resource_url,
-        'data-success_message' => options[:success_message]
+        'data-success_message' => options[:success_message],
+        'data-confirm_message' => options[:confirm_message]
```
Comment (mpapis) [inline lib/activeadmin_addons/addons/toggle_bool_builder.rb:28]:
```suggestion
        'data-confirm_message' => options[:confirm_message],
        'data-confirm_on_message' => options[:confirm_on_message],
        'data-confirm_off_message' => options[:confirm_off_message]
```

**Review by mpapis — COMMENTED** [review]
As I had use case to confirm only turning off - so extended with confirmation separately on or off.

File: app/javascript/activeadmin_addons/addons/toggle_bool.js [hunk 4]
```diff
@@ -2,6 +2,20 @@ var initializer = function() {
...
+    var confirmMessage = boolSwitch.data('confirm_on_message');
+    if (confirmMessage && !value && !confirm(confirmMessage)) {
+      return false;
+    }
+    var confirmMessage = boolSwitch.data('confirm_off_message');
+    if (confirmMessage && value && !confirm(confirmMessage)) {
+      return false;
+    }
+
```
[the remark this answers is not in the document]
Comment (mpapis) [reply app/javascript/activeadmin_addons/addons/toggle_bool.js:18]: I like it, but it has one down side - no ability to specify separate on/off messages at the same time, not an issue for me at this time tho.

[the remark this answers is not in the document]
Comment (rjherrera) [reply app/javascript/activeadmin_addons/addons/toggle_bool.js:18]: You are right, though I am inclined to think that with one message it'll be enough for most cases. 

I think we should proceed with this approach, and see if there is a need for separate messages and reevaluate 🙂

[the remark this answers is not in the document]
Comment (rjherrera) [reply app/javascript/activeadmin_addons/addons/toggle_bool.js:18]: @mario-amazing if you accept the changes we can merge!

[the remark this answers is not in the document]
Comment (rjherrera) [reply app/javascript/activeadmin_addons/addons/toggle_bool.js:18]: oh, we should also update the docs

File: app/javascript/activeadmin_addons/addons/toggle_bool.js [hunk 5]
```diff
@@ -2,6 +2,20 @@ var initializer = function() {
...
+if (confirmMessage) {
+  var shouldConfirm = 
+    (confirmTrigger === 'both') || 
+    (confirmTrigger === 'on' && !value) || 
+    (confirmTrigger === 'off' && value);
+
+  if (shouldConfirm && !confirm(confirmMessage)) {
+    return false;
+  }
+}
```
Comment (rjherrera) [inline app/javascript/activeadmin_addons/addons/toggle_bool.js:18]:
```suggestion
    var value = boolSwitch.data('value');
    var confirmMessage = boolSwitch.data('confirm_message');
    var confirmTrigger = boolSwitch.data('confirm_message_trigger') || 'both';
    
    if (confirmMessage) {
      var shouldConfirm = 
        (confirmTrigger === 'both') || 
        (confirmTrigger === 'on' && !value) || 
        (confirmTrigger === 'off' && value);
    
      if (shouldConfirm && !confirm(confirmMessage)) {
        return false;
      }
    }
```

File: docs/toggle_bool.md [hunk 6]
```diff
@@ -39,6 +39,19 @@ The value update is done through the default update route, so you must check you
...
+ Optionally, you can provide a confirmation message to the user before updating the value.
+  
+ This is disabled by default, but can be enabled by adding a option:
+ - `confirm_message` (on/off),
+ - `confirm_on_message` (on),
+ - `confirm_off_message` (off)
+   
+ `toggle_bool_column :paid, confirm_message: 'Are you sure you want to toggle this switch?'`
+  
+ If the user cancels the confirmation, the update action will be aborted.
```
Comment (rjherrera) [inline docs/toggle_bool.md:54]:
```suggestion
 ### Confirmation message

 Optionally, you can provide a confirmation message to the user before updating the value.
  
 This is disabled by default, but can be enabled by adding the `confirm_message` option. Optionally, you can add the `confirm_message_trigger` option to choose when to toggle the confirm message, possible values are:
 - `both` (default),
 - `on`
 - `off`
   
 `toggle_bool_column :paid, confirm_message: 'Are you sure you want to toggle this switch?', confirm_message_trigger: 'on'`
  
 If the user cancels the confirmation, the update action will be aborted.
```

File: lib/activeadmin_addons/addons/toggle_bool_builder.rb [hunk 7]
```diff
@@ -24,7 +24,10 @@ def toggle
         'data-field' => attribute,
         'data-value' => data,
         'data-url' => resource_url,
-        'data-success_message' => options[:success_message]
+        'data-success_message' => options[:success_message],
+        'data-confirm_message' => options[:confirm_message],
+        'data-confirm_on_message' => options[:confirm_on_message],
+        'data-confirm_off_message' => options[:confirm_off_message]
```
Comment (rjherrera) [inline lib/activeadmin_addons/addons/toggle_bool_builder.rb:30]:
```suggestion
        'data-confirm_message_trigger' => options[:confirm_message_trigger]
```

**Review by rjherrera — COMMENTED** [review]
i left the suggestions there

**Pull request merged**